recipes/mongo-cxx-driver/all/conanfile.py

```diff
@@ -148,15 +123,41 @@ def _configure_cmake(self):
         self._cmake.configure(build_folder=self._build_subfolder)
         return self._cmake
 
+    def validate(self):
+        if self.options.with_ssl and not bool(self.options["mongo-c-driver"].with_ssl):
+            raise ConanInvalidConfiguration("mongo-cxx-driver with_ssl=True requires mongo-c-driver with a ssl implementation")
+
+        if self.options.polyfill == "mnmlstc":
+            # TODO: add mnmlstc polyfill support
+            # Cannot model mnmlstc (not packaged, is pulled dynamically) polyfill dependencies
+            raise ConanInvalidConfiguration("mnmlstc polyfill is not yet supported")
+
+        if self.settings.compiler.get_safe("cppstd"):
+            tools.check_min_cppstd(self, self._minimal_std_version)
+
+        compiler = str(self.settings.compiler)
+        if self.options.polyfill == "experimental" and compiler == "apple-clang":
+            raise ConanInvalidConfiguration("experimental polyfill is not supported for apple-clang")
+
+        if compiler not in self._compilers_minimum_version:
+            self.output.warn("Unknown compiler, assuming it supports at least C++{}".format(self._minimal_std_version))
+            return
+
+        version = tools.Version(self.settings.compiler.version)
+        if version < self._compilers_minimum_version[compiler]:
+            raise ConanInvalidConfiguration(
+                "{} requires a compiler that supports at least C++{}".format(
```

Comment:
I'll have to read that blog post but I would have expected this in configure method